sycl/include/sycl/backend_types.hpp

```diff
@@ -65,5 +65,30 @@ inline std::ostream &operator<<(std::ostream &Out, backend be) {
   return Out;
 }
 
+namespace detail {
+inline std::string_view get_backend_name_no_vendor(backend Backend) {
```

Comment:
Seeing as this is not used anywhere except the sycl-ls tool, this should probably be moved to that source file.

Reply:
The reason I put it here is because any new entry to the enum has to come with similar updates both here and in `operator<<`. Putting it in a different file would require changes in multiple places where it would be easier to forget some.

Reply:
Alright, that makes sense.